Extensions/mutualchecker.js

```diff
@@ -0,0 +1,103 @@
+//* TITLE Mutual Checker **//
+//* VERSION 1.0.0 **//
+//* DESCRIPTION Adds an icon next to usernames to indicate if they follow you. **//
+//* DETAILS Does what it says on the tin! Made from fresh, distilled Profiler juice. **//
+//* DEVELOPER AprilSylph **//
+//* FRAME false **//
+//* BETA false **//
+
+XKit.extensions.mutualchecker = new Object({
+	
+	running: false,
+	
+	preferences: {
+		"main_blog": {
+			text: "Blog to check for follow-backs on:",
+			default: "",
+			value: "",
+			type: "blog"
+		},
+		"live_update": {
+			text: "Let me see users follow or unfollow me as I scroll",
+			default: false,
+			value: false,
+			slow: true
+		}
+	},
+	
+	run: function() {
+		this.running = true;
+		if (this.preferences.main_blog.value === "") { this.preferences.main_blog.value = XKit.tools.get_blogs()[0]; }
+		window.xkitmc = new Array();
+		XKit.tools.init_css("mutualchecker");
+		XKit.post_listener.add("mutualchecker", XKit.extensions.mutualchecker.add_post_icons);
+		XKit.extensions.mutualchecker.add_follower_icons();
+	},
+	
+	add_follower_icons: function() {
+		$(".follower").each(function() {
+			var m_url = $(this).find(".name").text();
+			$(this).find(".name").append("<span class=\"mutualcheck xkitmc-" + m_url + "\"></span>");
+			XKit.interface.is_following(m_url, XKit.extensions.mutualchecker.preferences.main_blog.value).then(function(follow) {
+				if (follow) {
+					XKit.extensions.mutualchecker.add_label(m_url);
+				}
+			});
+		});
+	},
+	
+	add_post_icons: function() {
+		var posts = XKit.interface.get_posts("mutualchecker-done");
+		
+		$(posts).each(function() {
+			$(this).addClass("mutualchecker-done");
+			var m_post = XKit.interface.post($(this));
+			if (XKit.interface.where().inbox !== true && m_post.is_mine === true) { return; }
+			var post_owner = m_post.owner;
+			
+			if (m_post.type === "note" && XKit.interface.where().inbox === true) {
+				var m_json_info = $(this).find(".post_avatar_link").attr('data-tumblelog-popover');
+				try {
+					var m_json_obj = JSON.parse(m_json_info);
+					post_owner = m_json_obj.name;
+				} catch (e) {
+					return;
+				}
+			}
+			
+			var mutual_indicator = "<span class=\"mutualcheck xkitmc-" + post_owner + "\"></span>";
+			var name_div = $(this).find(".post_info_fence").find("a").first();
+			name_div.after(mutual_indicator);
+			
+			if (typeof window.xkitmc[post_owner] === "undefined") {
```

Comment:
this looks like you're using it as an object rather then an array. maybe it should be an object?